### PR TITLE
dartsim: avoid revision bumps whenever dependencies' Cellar paths change

### PR DESCRIPTION
**Pull request by jslee02**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/dartsim.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Dartsim < Formula
   revision 1
 
   bottle do
+    rebuild 1
```
Comment (scpeters) [inline Formula/dartsim.rb]: I don't know if we're supposed to bump this. I thought I saw somewhere that we're just supposed to bump the formula revision

Comment (jslee02) [reply Formula/dartsim.rb]: Right, it seems bumping the formula revision is the right way to rebuild bottles per dependency updates (see: https://github.com/jslee02/homebrew-core/commit/5d3644c33079df05375d4b15a6ade8beccd50edf). Thanks @scpeters !

**Comment by ilovezfs** [comment]
Why is this needed? I think users can just `brew upgrade libccd`, no?

**Comment by jslee02** [comment]
It seems `dartsim` includes a path to an old version of `libccd`, which is configured in CMake run (see: https://travis-ci.org/personalrobotics/aikido/jobs/398432647#L5262-L5274). Hm, we maybe need to fix `dartsim` not to include the path?

**Comment by ilovezfs** [comment]
I see.

```
./share/dart/cmake/dart_dartTargets.cmake:  INTERFACE_INCLUDE_DIRECTORIES "/usr/local/include/eigen3;/usr/local/Cellar/libccd/2.0_2/include;/usr/local/include;/usr/local/include;/usr/local/include"
./share/dart/cmake/dart_dartTargets.cmake:  INTERFACE_LINK_LIBRARIES "/usr/local/Cellar/libccd/2.0_2/lib/libccd.dylib;/usr/local/Cellar/fcl/0.5.0_1/lib/libfcl.dylib;/usr/local/lib/libassimp.dylib;/usr/local/lib/libboost_regex-mt.dylib;/usr/local/lib/libboost_system-mt.dylib;/usr/local/lib/libboost_filesystem-mt.dylib;dart-external-odelcpsolver"
./share/dart/cmake/dart_dartTargets.cmake:  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "/usr/local/include/eigen3;/usr/local/Cellar/libccd/2.0_2/include;/usr/local/include;/usr/local/include;/usr/local/include"
```

I guess we could rewrite all of those Cellar paths as opt paths.

**Comment by ilovezfs** [comment]
e.g. https://github.com/Homebrew/homebrew-core/blob/master/Formula/octave.rb#L82-L86

**Comment by jslee02** [comment]
Thanks for the reference @ilovezfs. Let me update the `dartsim` formula accordingly tomorrow. Feel free to close this PR.

**Comment by ilovezfs** [comment]
You can just force push to this one if you want.

**Comment by jslee02** [comment]
@ilovezfs I updated according to your suggestion. Is there a smarter way to specify the file path (`dart_dartTargets.cmake`) or files (`*Targets.cmake`) instead of using the exact relative path, which is rather brittle? :thinking:

**Comment by scpeters** [comment]
there's another Cellar reference to urdfdom:

~~~
share/dart/cmake/dart_utils-urdfTargets.cmake:57:  INTERFACE_INCLUDE_DIRECTORIES "/usr/local/Cellar/urdfdom/1.0.0_1/include;/usr/local/include;/usr/local/Cellar/urdfdom_headers/1.0.0/include;/usr/local/include"
share/dart/cmake/dart_utils-urdfTargets.cmake:58:  INTERFACE_LINK_LIBRARIES "dart-utils;/usr/local/Cellar/urdfdom/1.0.0_1/lib/liburdfdom_sensor.dylib;/usr/local/Cellar/urdfdom/1.0.0_1/lib/liburdfdom_model_state.dylib;/usr/local/Cellar/urdfdom/1.0.0_1/lib/liburdfdom_model.dylib;/usr/local/Cellar/urdfdom/1.0.0_1/lib/liburdfdom_world.dylib;console_bridge"
share/dart/cmake/dart_utils-urdfTargets.cmake:59:  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "/usr/local/Cellar/urdfdom/1.0.0_1/include;/usr/local/include;/usr/local/Cellar/urdfdom_headers/1.0.0/include;/usr/local/include"
~~~

**Comment by scpeters** [comment]
I don't know if there's a less brittle way of doing this, but it would be interesting if there was some kind of `brew audit` that checked bottle contents for references to the cellar.

**Comment by jslee02** [comment]
Thanks, @scpeters. Added `urdfdom` and `urdfdom_headers` to the list.

**Review by scpeters — APPROVED** [review]

**Comment by jslee02** [comment]
This is a gentle reminder. @ilovezfs Does this PR look good to you?

**Comment by LorenzMeier** [comment]
@ilovezfs @scpeters Merging it would be much appreciated to fix a widely used robotics simulator. Thanks!

**Comment by scpeters** [comment]
@LorenzMeier thanks for the feedback; I've already approved these changes, so it's just up to a homebrew maintainer at this point

**Comment by ilovezfs** [comment]
Interestingly, locally, notably on a case sensitive file system, I get
```
/tmp/dartsim-20180714-68253-tfsyfr/dart-6.5.0/dart/gui/LoadGlut.hpp:42:12: fatal error: 'Glut/glut.h' file not found
  #include <Glut/glut.h>
           ^
1 error generated.
make[2]: *** [dart/gui/CMakeFiles/dart-gui.dir/GLFuncs.cpp.o] Error 1
make[1]: *** [dart/gui/CMakeFiles/dart-gui.dir/all] Error 2
make: *** [all] Error 2
```

Odd given the formula depends on freeglut which has its glut.h at `<GL/glut.h>`
```
bash-3.2$ ls /usr/local/opt/freeglut/include/GL/
freeglut.h	freeglut_ext.h	freeglut_std.h	glut.h
bash-3.2$ 
```

Also odd the formula depends on freeglut at all given macOS has /System/Library/Frameworks/GLUT.framework

**Comment by ilovezfs** [comment]
>Is there a smarter way to specify the file path (dart_dartTargets.cmake) or files (*Targets.cmake) instead of using the exact relative path, which is rather brittle?

@jslee02 yes I've changed it to use `share/"dart/cmake/dart_dartTargets.cmake"` and `share/"dart/cmake/dart_utils-urdfTargets.cmake"` after `make install` has already run, so that we don't have to know the path in the source tree.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks again @jslee02! Shipped.

**Pull request closed without merging**